File: python/enolib/elements/attribute.py
from . import field
from .value_element_base import ValueElementBase


class Attribute(ValueElementBase):
    def __repr__(self):
        return f"<class Attribute key={self._instruction['key']} value={self._print_value()}>"

    def parent(self):
        return self._parent or field.Field(self._context, self._instruction.parent)

File: ruby/lib/enolib/elements/attribute.rb
# frozen_string_literal: true

module Enolib
  class Attribute < ValueElementBase
    attr_reader :instruction # TODO: Revisit this hacky exposition
    
    def parent
      @parent || Field.new(@context, @instruction[:parent])
    end

    def to_s
      "#<Enolib::Attribute key=#{@instruction[:key]} value=#{print_value}>"
    end
  end
end
